plugins/PrivacyManager/PrivacyManager.php

```diff
@@ -741,4 +744,42 @@ private function shouldRenderFooterLinks(SystemSettings $settings)
 
         return false;
     }
+
+    public function shouldAddTrackerFile(&$shouldAdd, $pluginName)
+    {
+        if ($pluginName === 'PrivacyManager') {
+            // only check the config value, so no database connection needs to be established
+            $shouldAdd = !!PiwikConfig::getInstance()->Tracker['force_cookieless_tracking'];
+        }
+    }
+
+    /**
+     * Returns if cookie less tracking is forced globally or for the given site
+     *
+     * @param null|int $idSite
+     * @return bool
+     */
+    public static function isCookieLessTrackingForced($idSite=null)
+    {
+        try {
+            if (PiwikConfig::getInstance()->Tracker['force_cookieless_tracking']) {
```

Comment:
here we would also use the system setting instead of the config (it should be cached in tracker mode automatically)

Reply:
I see the result is cached for tracking anyway so it doesn't need to be cached anyway in the settings class directly